core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -537,7 +537,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
 
     _executorAllocationManager =
-      if (dynamicAllocationEnabled) {
+      if (dynamicAllocationEnabled &&
+        (!isLocal || conf.getBoolean("spark.dynamicAllocation.testing", false))) {
```

Comment:
Make sense. Update the PR to move it into Utils.isDynamicAllocationEnabled and have some refactoring accordingly.